lib/services/account-api-service.js

```diff
@@ -59,17 +59,25 @@ function accountApiServiceFactory(
             return Promise.all(promises);
         }).then(function() {
             // Enforce a subset of hardcoded privileges:
+            var operator = {
+                role: 'Operator',
+                privileges: ['Login', 'ConfigureComponents', 'ConfigureSelf']
+            };
+            var admin = {
+                role: 'Administrator',
+                privileges: ['Read', 'Write', 'Login', 'ConfigureManager', 'ConfigureUsers', 'ConfigureComponents', 'ConfigureSelf']
+            };
+            var readOnly = {
+                role: 'ReadOnly',
+                privileges: ['Read', 'Login', 'ConfigureSelf']
+            }
             return Promise.all([
-                // 2.0 API's fixed privileges
-                acl.addRoleParents('Administrator', ['Read', 'Write']),
-                acl.addRoleParents('ReadOnly', ['Read']),
-
-                // Redfish 1.0 fixed privileges
-                acl.addRoleParents('Administrator', ['Login','ConfigureManager','ConfigureUsers']),
-                acl.addRoleParents('Administrator', ['ConfigureComponents','ConfigureSelf']),
-                acl.addRoleParents('Operator', ['Login','ConfigureComponents','ConfigureSelf']),
-                acl.addRoleParents('ReadOnly', ['Login', 'ConfigureSelf'])
-            ]);
+                self.createRole(readOnly),
+                self.createRole(operator),
+                self.createRole(admin)
+            ]).catch(function () {
+                //mask error here
```

Comment:
@yyscamper this is done on purpose to mask the error. This function needs to run on startup. This part will create the essential roles (Administrator, ReadOnly and operator) if they don't exist